service/sqs/checksums.go

```diff
@@ -74,7 +74,9 @@ func verifyReceiveMessage(r *request.Request) {
 		for _, msg := range out.Messages {
 			err := checksumsMatch(msg.Body, msg.MD5OfBody)
 			if err != nil {
-				ids = append(ids, *msg.MessageId)
+				if msg.MessageId != nil {
```

Comment:
We probably want to log the fact that a `MessageId` was not present in the response. In addition the `RequestID` from `request.Request` should be included in the message so that the user is aware of the issue.